src/librustc_resolve/diagnostics.rs

```diff
@@ -964,6 +964,31 @@ impl<'a> Resolver<'a> {
 
         err.emit();
     }
+
+    crate fn make_undeclared_type_suggestion(
+        &mut self,
+        ident: Ident,
+    ) -> (String, Option<Suggestion>) {
+        let parent_scope = &ParentScope::module(self.graph_root);
+        let typo_suggestion = self.early_lookup_typo_candidate(
+            ScopeSet::AbsolutePath(TypeNS),
+            parent_scope,
+            ident,
+            &|_| true,
+        );
+        if let Some(typo_suggestion) = typo_suggestion {
```

Comment:
Could you copypaste the wording from `fn add_typo_suggestion`?
Or break `add_typo_suggestion` into two functions, one providing the wording and another applying that wording as a suggestion / label.